plugins/Disintegrator/Disintegrator.cpp

```diff
@@ -0,0 +1,279 @@
+/*
+ * Disintegrator.cpp
+ *
+ * Copyright (c) 2019 Lost Robot <r94231@gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "Disintegrator.h"
+
+#include "embed.h"
+#include "lmms_math.h"
+#include "plugin_export.h"
+
+extern "C"
+{
+
+Plugin::Descriptor PLUGIN_EXPORT disintegrator_plugin_descriptor =
+{
+	STRINGIFY(PLUGIN_NAME),
+	"Disintegrator",
+	QT_TRANSLATE_NOOP("pluginBrowser", "A delay modulation effect for creating very aggressive digital distortion."),
+	"Lost Robot <r94231@gmail.com>",
+	0x0100,
+	Plugin::Effect,
+	new PluginPixmapLoader("logo"),
+	NULL,
+	NULL
+} ;
+
+}
+
+
+
+DisintegratorEffect::DisintegratorEffect(Model* parent, const Descriptor::SubPluginFeatures::Key* key) :
+	Effect(&disintegrator_plugin_descriptor, parent, key),
+	m_disintegratorControls(this)
+{
+	for (int a = 0; a < 200; ++a)
+	{
+		for (int b = 0; b < 2; ++b)
+		{
+			m_inBuf[b].push_back(0);
+		}
+	}
```

Comment:
This will run slightly faster:
```cpp
		for (int i = 0; i < 2; ++i)
		{
			m_inBuf[i].reserve(200);
			for (int j = 0; j < 200; ++j)
			{
				m_inBuf[i][j] = 0.0f;
			}
		}
```
Also, could you use a constant whose value is 200 instead of using the value directly? This will make things much easier if you change the value. Also, it will make clear what 200 means.

FYI, loop variables *usually* start from `i`, not `a`.

Reply:
Oh, I think this will be better: https://stackoverflow.com/a/34683195